MyApplication/app/src/main/java/com/example/myapplication/ui/main/viewmodel/MainViewModel.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.myapplication.ui.main.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.liveData
+import com.example.myapplication.data.repo.MainRepository
+import com.example.myapplication.utils.Resource
+import kotlinx.coroutines.Dispatchers
+import java.lang.Exception
+
+class MainViewModel(private val mainRepository: MainRepository) : ViewModel() {
+
+    fun getUsers() = liveData(Dispatchers.IO) {
+        emit(Resource.loading(data = null))
+        try {
+            emit(Resource.success(data = mainRepository.getUsers()))
+        } catch (exception: Exception) {
+            emit(Resource.error(data = null, message = exception.message ?: "Error Occurred!"))
+        }
+    }
+}
```

Comment:
@Dynna familiarize yourself with `ViewModel` testing before starting lab5. For [example](https://blog.mindorks.com/unit-testing-viewmodel-with-kotlin-coroutines-and-livedata)